kernel/priority_queues.c

```diff
@@ -94,16 +94,18 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq)
 
 struct k_thread *z_priq_mq_best(struct _priq_mq *pq)
 {
-	if (!pq->bitmask) {
-		return NULL;
-	}
+	for (int i = 0; i < K_NUM_THRAD_PRIO; ++i) {
+		if (!pq->bitmask[i]) {
+			continue;
+		}
 
-	struct k_thread *thread = NULL;
-	sys_dlist_t *l = &pq->queues[__builtin_ctz(pq->bitmask)];
-	sys_dnode_t *n = sys_dlist_peek_head(l);
+		sys_dlist_t *l = &pq->queues[__builtin_ctz(pq->bitmask[i])];
```

Comment:
I don't think that we are indexing the correct queue. There does not seem to be any extra addition to account for a skipped bitmask[i]. That is, if bitmask[0] is 0, then bitmask[1] would (incorrectly) index to the same queues that bitmask[0] would.